satpy/readers/generic_image.py

```diff
@@ -75,14 +84,9 @@ def read(self):
         # Rename bands to [R, G, B, A], or a subset of those
         data['bands'] = BANDS[data.bands.size]
 
-        # Mask data if alpha channel is present
-        try:
-            data = mask_image_data(data)
-        except ValueError as err:
-            logger.warning(err)
-
         data.attrs = attrs
-        self.file_content['image'] = data
+        self.dataset_name = 'image'
```

Comment:
Doesn't this mean that this will *always* be set to `"image"` which means the `ds_name = self.dataset_name if self.dataset_name else key['name']` on line 109 will always be `"image"`?

Reply:
Yes, absolutetly. This should just address the testing think discussed before:

> Regarding file_content, that is likely just a common practice used from other readers. It also makes writing tests easier, at least in other readers, because complex NetCDF/HDF files can be described by a dictionary and put in file_content. For this reader, maybe that isn't necessary

Someone can inherit this reader and override the ctor() and/or read() to provide own test data.

Reply:
But for your use case, where you want the YAML to configure a new dataset name, this pull request no longer handles that case right? This `self.dataset_name` is always set to `'image'` and the `get_dataset` method never actually uses `key['name']` to overwrite the `.attrs['name']`, right? Actually it looks like `get_dataset` never uses the provided `info` dictionary to update the metadata of the image (`.attrs` should include the information provided in `info`).

Does that make sense? I guess I'm just confused at how this works because at first glance it doesn't seem like it should.

Reply:
Hmm, I don't really know what for `attrs['name']` is used in satpy, but in my case it obviously doesn't matter. The problem was that in `get_dataset()` the access to `file_content` was with a different key than the data was put into it. And that is solved here (key is always `image`).

Reply:
@ch-k If you load a `Scene` using this reader, what does `print(scn.keys())` show you? The `.attrs['name']` is used by Satpy to actually provide the name that Satpy uses. Satpy is actually using `name` and other pieces of metadata to create a `DataID` object to fully/uniquely identify a DataArray. If the `.attrs` for the `DataArray` don't include what the YAML has then it is likely that multiple datasets will be loaded as a single item. Based on what I understand of this code, even if your YAML said it had "image1" and "image2", they would both be returned with "image" as their name.

It is possible the base reader is overwriting the attributes with the information from the YAML so that this doesn't matter, but I'd prefer the readers be explicit if possible. Sorry, if this doesn't make much sense.

Reply:
```
scn.load(['image_goes16', 'image_goes17'])
print(scn.keys())
```
> [DataID(name='image_goes16', modifiers=()), DataID(name='image_goes17', modifiers=())]

and after loading the derived composite:

```
scn.load(['dwd_world_comp'])
print(scn.keys())
```
> [DataID(name='dwd_world_comp'), DataID(name='image_goes16', modifiers=()), DataID(name='image_goes17', modifiers=())]

Looks fine.

Reply:
I digged a little bit more in satpy code. It seems that the DataID is derived from the reader yaml. This id is used to call the readers `get_dataset()` method and this id is also used as key to fill a DatasetDict (in yaml_reader.py `load()`). Couldn't really find a reader that explicitly sets `attrs['name']` (Just want to see how it should work).

Reply:
What you should find is file handlers that do `.attrs.update(info)` or something like that (ex. `DataArray(..., attrs=new_attrs)`). The `info` passed to the file handler is meant to be a dictionary of the info provided in the YAML file including DataID-like parameters like `name`, `calibration`, etc.

Reply:
Alright. Just pushed this change.